### ISSUE TITLE
is sbt-javacv dead ?

### ISSUE THREAD

**Issue by aashishamrute** [bytedeco/javacv#1513]
Hi,

is sbt-javacv dead ? if yes, is there any alternative. Currently including javacv-platform include all modules for all platform. is there workaround in SBT to include only OS dependent lib. Also,  I need Javacv with opencv + ffmpeg, I tried with exclude config but it do not exclude other jars.

Thanks

**Comment by saudet** [comment]
@jpsacha What do you recommend?

**Comment by jpsacha** [comment]
@aashishamrute You can see examples of using OpenCV with plain SBT here:
https://github.com/ij-plugins/ijp-javacv/blob/master/build.sbt
and here
https://github.com/bytedeco/javacv-examples/blob/master/OpenCV_Cookbook/build.sbt

You can selectively exclude transitive dependencies that do not need using normal SBT mechanisms. Details are here
https://www.scala-sbt.org/1.x/docs/Library-Management.html#Exclude+Transitive+Dependencies

**Comment by saudet** [comment]
@jpsacha It looks like a lot of code just to get started. It would be nice if you could put that logic in sbt-javacv. What do you think?

**Comment by jpsacha** [comment]
@saudet I am not clear what you are asking for. I am not very familiar with `sbt-javacv`. Judging by the README `sbt-javacpp` can be used to get just the required preset for a specific platform. You could use something like
```
javaCppVersion := "1.5.3"
javaCppPresetLibs ++= Seq("opencv" -> "4.3.0", "ffmpeg" -> "4.2.2", "openblas" -> "0.3.9")
```
It will only use those dependencies for the current platform. It should not load all the platforms.

In the link in previous answer I provided an alternative to `sbt-javacv` using vanilla SBT as the original question asked for an altenative.

**Comment by saudet** [comment]
I mean, could you move what you have in javacv-examples to sbt-javacv, and use sbt-javacv in javacv-examples instead?

**Comment by jpsacha** [comment]
`javacv-examples` keep up with latest releases of `javacpp` and dependent libraries including latest 1.5.4.
`sbt-javacv` only works till version 1.4.4, see issue #1194

**Comment by aashishamrute** [comment]
@saudet , @jpsacha thanks for document and link you provided. for now able to filter lib and platform specific lib, using [sbt-javacpp](https://github.com/bytedeco/sbt-javacpp). 

For now, I did the following.

Added sbt-javacpp plugin to plugins.sbt
`addSbtPlugin("org.bytedeco" % "sbt-javacpp" % "1.16")`

In build.sbt, 
`javaCppVersion := "1.5.4"

javaCppPresetLibs ++= Seq(
  "opencv" -> "4.4.0",
  "ffmpeg" -> "4.3.1",
  "openblas" -> "0.3.10"
)`
 
for JavaCV,
`"org.bytedeco" % "javacv"  % javaCVVersion
    exclude("org.bytedeco", "artoolkitplus")
    exclude("org.bytedeco", "flandmark")
    exclude("org.bytedeco", "tesseract")
    exclude("org.bytedeco", "libdc1394")
    exclude("org.bytedeco", "libfreenect")
    exclude("org.bytedeco", "libfreenect2")
    exclude("org.bytedeco", "librealsense")
    exclude("org.bytedeco", "librealsense2")
    exclude("org.bytedeco", "videoinput")
    exclude("org.bytedeco", "flycapture")
    exclude("org.bytedeco", "leptonica")`

one thing I notice, sbt-javacpp do not include javacpp platform specific lib, so I have to add this extra in build.sbt
`libraryDependencies ++= "org.bytedeco" % "javacpp"  % javaCVVersion classifier Loader.getPlatform`

I will followup on sbt-javacv update with issue #1194
